Kernel/Graphics/FramebufferDevice.cpp

```diff
@@ -113,6 +119,7 @@ UNMAP_AFTER_INIT void FramebufferDevice::initialize()
 
 UNMAP_AFTER_INIT FramebufferDevice::FramebufferDevice(const GraphicsDevice& adapter, size_t output_port_index, PhysicalAddress addr, size_t width, size_t height, size_t pitch)
     : BlockDevice(29, GraphicsManagement::the().allocate_minor_device_number())
+    , m_device_id(allocate_device_id())
```

Comment:
As @IdanHo indicated, how this is different from `GraphicsManagement::the().allocate_minor_device_number()` we are using right now?
Either we keep the `GraphicsManagement::the().allocate_minor_device_number()` solution or change it, but don't do both. 
Personally, I prefer my solution, because it seems correct to allocate the minor device number from the Singleton, so we can do other locking if required (hotplug events, other events of hardware that I don't have example for now).

Reply:
Ah, I believe that m_device_id is the framebuffer number (e.g. `/dev/fb0` -> m_device_id = 0). I believe this is distinct from allocate_minor_device(), since the framebuffer numbers need to be contiguous and starting from 0, which might not be true for the allocated minor device numbers if multiple display adaptors are present.

Reply:
these are allocated the exact same number of times, at the same locations, and start with the same initial value, so there is no difference between them as far as i can tell.

Reply:
Yes, but `GraphicsManagement::the().allocate_minor_device_number()` can be called by other devices in between. If other devices get allocated, then they will get out of sync.

Reply:
`FrameBuffer::allocate_device_id()` Can be called by the framebuffers of other devices as well, no?

Reply:
Fair point.

Reply:
Yeah this is indeed no longer needed. I dropped the commit and it works just fine :)